catalog/controller/extension/payment/razorpay.php

```diff
@@ -350,4 +391,105 @@ protected function getApiIntance()
         return new Api($this->config->get('payment_razorpay_key_id'), $this->config->get('payment_razorpay_key_secret'));
     }
 
+    protected function autoCreateWebhook()
+    {
+        $this->webhookSecret = $this->config->get('payment_razorpay_webhook_secret');
+
+        if(empty($this->webhookSecret) === true)
+        {
+            $this->webhookSecret = bin2hex(openssl_random_pseudo_bytes(4));
+        }
+
+        $api = $this->getApiIntance();
+
+        $domain = parse_url($this->webhookUrl, PHP_URL_HOST);
+        $domain_ip = gethostbyname($domain);
+
+        if (!filter_var($domain_ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4 | FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE))
+        {
+            $this->webhookEnable = '0';
+            $this->log->write('Cannot enable/disable webhook on $domain or private ip($domain_ip).');
+            return;
+        }
+
+        try
+        {
+            $webhookPresent = $this->getExistingWebhook();
+
+            if(empty($this->webhookId) === false)
+            {
+                $webhook = $api->webhook->edit(
+                    [
+                        'url'    => $this->webhookUrl,
+                        'events' => $this->webhookEvents,
+                        'secret' => $this->webhookSecret,
+                        'active' => true,
+                    ],
+                    $this->webhookId
+                );
+
+                $this->log->write('Razorpay Webhook Updated by Admin.');
+            } else
+            {
+                $webhook = $api->webhook->create(
+                    [
+                        'url'    => $this->webhookUrl,
+                        'events' => $this->webhookEvents,
+                        'secret' => $this->webhookSecret,
+                        'active' => true,
+                    ]
+                );
+
+                $this->log->write('Razorpay Webhook Created by Admin');
+            }
+
+            $this->webhookEnable = '1';
+        }
+        catch(\Razorpay\Api\Errors\Error $e)
+        {
+            $this->webhookEnable = '0';
+
+            $this->log->write($e->getMessage());
+        }
+    }
+
+    protected function getExistingWebhook()
+    {
+        $api = $this->getApiIntance();
+
+        try
+        {
+            $webhooks = $api->webhook->all();
+
+            if(($webhooks->count > 0) and
+                (empty($this->webhookUrl) === false))
+            {
+                foreach ($webhooks->items as $key => $webhook)
+                {
+                    if($webhook->url === $this->webhookUrl)
+                    {
+                        $this->webhookId = $webhook->id;
+
+                        foreach ($webhook->events as $event => $status)
+                        {
+                            if($status === true and in_array($event,$this->webhookSupportedEvents))
```

Comment:
new condition should be start in new line

Reply:
space after comma

Reply:
@ramth05 updated code with space after comma and new condition in new line